user/-j-andrew/test&automation-edit (#2431)
MicrosoftDocs/edge-developer: microsoft-edge/test-and-automation/test-and-automation.md

| Playwright | The Playwright library provides cross-browser automation through a single API. |
| Puppeteer | The Puppeteer library provides a high-level API to control browsers, including Microsoft Edge, using the DevTools Protocol. |
| WebDriver | Automates testing in Microsoft Edge by simulating user interaction.  Provides advantages over JavaScript unit tests. |

> | Webhint | Checks your code for best practices and common errors, to test and improve accessibility, performance, cross-browser and PWA compatibility, and security of your site. |

Reviewer: ```suggestion
| webhint | Checks your code for best practices and common errors, to test and improve accessibility, performance, cross-browser and PWA compatibility, and security of your site. |
```
webhint is intentionally not capitalized. See https://webhint.io/

<!-- ====================================================================== -->
## DevTools Protocol

> Use the DevTools Protocol to instrument, inspect, debug, and profile browsers, including Microsoft Edge. By [building Microsoft Edge on the Chromium open-source project](https://github.com/MicrosoftEdge/MSEdge/blob/master/README.md), the Microsoft Edge DevTools Protocol matches the APIs of the Chrome DevTools Protocol.

Reviewer 2: ```suggestion
Use the DevTools Protocol to instrument, inspect, debug, and profile browsers, including Microsoft Edge.  By building Microsoft Edge on the Chromium open-source project, the Microsoft Edge DevTools Protocol matches the APIs of the Chrome DevTools Protocol.  For information about the Chromium open-source project, see [Microsoft Edge and Chromium Open Source: Our Intent](https://github.com/MicrosoftEdge/MSEdge/blob/master/README.md).
```
instance 2 of the above comment

You can use Origin Trials to try out experimental APIs on live sites for a limited period of time.  When using Origin Trials, users of Microsoft Edge that visit your site may run code that uses experimental APIs.  To access the experimental APIs on each user machine, you don't need to go to `edge://flags` and turn on feature flags.

> See the [Microsft Edge Origin Trials Developer Console](https://developer.microsoft.com/en-us/microsoft-edge/origin-trials/) to see a list of the available origin trials and register your origin to give them a try.  You can also provide feedback on the design of the API, your use cases, or your experience using the APIs to browser engineers and the web standard community.

Reviewer 2: ```suggestion
To see a list of the available origin trials and register your origin to give them a try, see the [Microsft Edge Origin Trials Developer Console](https://developer.microsoft.com/microsoft-edge/origin-trials/).  You can also provide feedback to browser engineers and the web standard community about the design of the API, your use cases, or your experience using the APIs.
```
